.hooks/pre-commit

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+#
+CXX_FORMAT="clang-format"
+CXX_EXTS=".c .h .cpp .hpp .cc .hh .cxx .hxx"
+
+PY_FORMAT="autopep8"
+PY_EXTS=".py"
+
+command_exists() {
+	hash $1 2>/dev/null
+}
+
+match_extension() {
+	local filename=$(basename $1)
+	local extension=".${filename##*.}"
+	local ext
+
+	for ext in $2; do [[ "$ext" == "$extension" ]] && return 0; done
+	return 1
+}
+
+if ! command_exists $CXX_FORMAT; then
+	printf "Error: $CXX_FORMAT executable not found.\n"
+	exit 1
+fi
+
+if ! command_exists $PY_FORMAT; then
+	printf "Error: $PY_FORMAT executable not found.\n"
+	exit 1
```

Comment:
I am not sure whether we should give one a prompt to ask if they want to continue nontheless...